frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/EditContactCard.tsx

```diff
@@ -0,0 +1,399 @@
+import React, { useState, useMemo } from "react";
+import {
+  Box,
+  Button,
+  Divider,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Heading,
+  Input,
+  Spacer,
+  Text,
+  Textarea,
+  useDisclosure,
+  useToast,
+  Wrap,
+  WrapItem,
+} from "@chakra-ui/react";
+import {
+  PersonalInfoActions,
+  PersonalInfoReducerDispatch,
+} from "../../../../../types/PersonalInfoTypes";
+import { EmergencyContact } from "../../../../../types/CamperTypes";
+import RequiredAsterisk from "../../../../common/RequiredAsterisk";
+import DeleteModal from "../../../../common/DeleteModal";
+import {
+  checkEmail,
+  checkFirstName,
+  checkLastName,
+  checkPhoneNumber,
+  checkRelationToCamper,
+} from "../PersonalInfo/personalInfoReducer";
+
+type EditContactCardProps = {
+  contact: EmergencyContact;
+  contactIndex: number;
+  dispatchPersonalInfoAction: (action: PersonalInfoReducerDispatch) => void;
+};
+
+const EditContactCard = ({
+  contact,
+  contactIndex,
+  dispatchPersonalInfoAction,
+}: EditContactCardProps): React.ReactElement => {
+  const [updateMemo, setUpdateMemo] = useState(0);
+  const initialContact: EmergencyContact = useMemo(() => contact, [updateMemo]);
+
+  const [editing, setEditing] = useState(false);
+
+  const [isFirstNameInvalid, setIsFirstNameInvalid] = useState<boolean>(false);
+  const [isLastNameInvalid, setIsLastNameInvalid] = useState<boolean>(false);
+  const [isEmailInvalid, setIsEmailInvalid] = useState<boolean>(false);
+  const [isPhoneNumberInvalid, setIsPhoneNumberInvalid] = useState<boolean>(
+    false,
+  );
+  const [isRelationInvalid, setIsRelationInvalid] = useState<boolean>(false);
+
+  const updateFormErrorMsgs = () => {
+    // Check if we're on secondary contact. If so, then we display error messages only if user has started filling secondary contact.
+    if (
+      contactIndex === 1 &&
+      !(
+        contact.firstName ||
+        contact.lastName ||
+        contact.email ||
+        contact.phoneNumber ||
+        contact.relationshipToCamper
+      )
+    ) {
+      setIsFirstNameInvalid(false);
+      setIsLastNameInvalid(false);
+      setIsEmailInvalid(false);
+      setIsPhoneNumberInvalid(false);
+      setIsRelationInvalid(false);
+      return;
+    }
+
+    let valid = true;
+
+    if (!checkFirstName(contact.firstName)) {
+      setIsFirstNameInvalid(true);
+      valid = false;
+    }
+    if (!checkLastName(contact.lastName)) {
+      setIsLastNameInvalid(true);
+      valid = false;
+    }
+    if (!checkEmail(contact.email)) {
+      setIsEmailInvalid(true);
+      valid = false;
+    }
+    if (!checkPhoneNumber(contact.phoneNumber)) {
+      setIsPhoneNumberInvalid(true);
+      valid = false;
+    }
+    if (!checkRelationToCamper(contact.relationshipToCamper)) {
+      setIsRelationInvalid(true);
+      valid = false;
+    }
+    if (valid) {
+      setEditing(false);
+      setUpdateMemo(updateMemo + 1);
+    }
+  };
+
+  const CancelChangesModal = () => {
```

Comment:
same as the other Edit card, thoughts on making an Edit card footer component which you can pass in these onDelete and other details as props?

Reply:
Done, as above.